docs/glossary.md

```diff
@@ -0,0 +1,161 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a branch that doesn't have any [branches](#branch)
+pointing to it or to any of its descendants. Unlike Git, Jujutsu keeps such
+commits around until they are explicitly abandoned.
+
+## Backend
+
+A backend is an implementation of the storage layer. When unqualified, we
```

Comment:
A backend is an implementation of the storage layer. Currently, the setup jj usually uses is to use a Git repository to store commit contents (this is the "commit backend") and a custom backend to store [the operation log](#operation-log). Another bare-bones commit backend exists for testing purposes only.

Alternative backends could be used, for example, if somebody wanted to use `jj` with a humongous monorepo.

Reply:
Thanks. I copied some parts and rewrote other parts. I feel like "custom backend" is equivalent to "out-of-tree backend", which is not what we mean here.